src/Stratis.Bitcoin.Features.PoA/PoAMiner.cs

```diff
@@ -99,7 +99,10 @@ private async Task CreateBlocksAsync()
 
                     // TODO sign the block with our signature
 
-                    ChainedHeader chainedHeader = await this.consensusManager.BlockMinedAsync(blockTemplate.Block).ConfigureAwait(false);
+                    // Update merkle root.
+                    blockTemplate.Block.UpdateMerkleRoot();
```

Comment:
Why?

Reply:
Because without it the root is 0000000000000000000 and when I sync other node it fails at merkle tree root integrity validaton rule

Reply:
but this should happen already somewhere else in the processes no?

Reply:
No, we need to do it manually when we are done modifying the block.